File: TopInterview150/Hashmap/RansomNote/solution.py
class Solution(object):
    def canConstruct(self, ransomNote, magazine):
        """
        :type ransomNote: str
        :type magazine: str
        :rtype: bool
        """
        count = {}
        for c in magazine:
            if c in count:
                count[c]+=1
            else:
                count[c] = 1
        for c in ransomNote:
            if c not in count or count[c]==0:
                return False
            count[c]-=1
        return True

File: TopInterview150/Hashmap/RansomNote/solution.cpp
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        unordered_map<char, int> magazineCount;
        for (char c : magazine)
        {
            if (magazineCount.find(c) == magazineCount.end())
            {
                magazineCount.insert({c, 1});
            }
            else magazineCount[c]++;
        }
        for (char c: ransomNote)
        {
            if (magazineCount.find(c) == magazineCount.end() || magazineCount[c] == 0)
            {
                return false;
            }
            magazineCount[c]--;
        }
        return true;
    }
};